### PR TITLE
set Repository ID

### PR DESCRIPTION
**Pull request by nevillelyh**
`Repository` checks `getId` for equality http://grepcode.com/file/repo1.maven.org/maven2/org.apache.maven/maven-model/3.3.3/org/apache/maven/model/RepositoryBase.java#RepositoryBase.equals%28java.lang.Object%29

Without this, the `toSet` always end up with 1 repo https://github.com/bazelbuild/migration-tooling/pull/76/files#diff-4518b099acd871fca1a03fff14b3df1bL88

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

**Comment by nevillelyh** [comment]
I signed it!

**Comment by nevillelyh** [comment]
Just realized that this fixes the same thing as #74 but I believe this is a cleaner fix.

**Comment by nevillelyh** [comment]
Is this still maintained? @kchodorow please take a look maybe?

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

**Comment by ulfjack** [comment]
@aehlig @dslomov

**Comment by ulfjack** [comment]
I'll merge this if nobody objects in the next two days.

**Comment by vorburger** [comment]
@ulfjack did you mean "in the next two = 2 or 20 days" ? :smiley_cat:

**Review by ulfjack — APPROVED** [review]

**Review by ulfjack — APPROVED** [review]

**Comment by ulfjack** [comment]
Apparently I can't merge it. :-/

**Pull request merged**
